src/DependencyInjection/Configuration.php

```diff
@@ -34,7 +34,7 @@ public function getConfigTreeBuilder()
                     ->addDefaultsIfNotSet()
                     ->children()
                         ->booleanNode('builder_alias')->defaultTrue()->end()
-                        ->booleanNode('container_aware')->defaultTrue()->end()
+                        ->booleanNode('container_aware')->setDeprecated('The "%node%" option is deprecated and will be removed in version 3.')->defaultTrue()->end()
```

Comment:
is this feature available in all our target symfony/config versions ? Or do we need to trigger the deprecation manually in `beforeNormalization` ?

Reply:
Feature has been [introduced in 3.4](https://symfony.com/blog/new-in-symfony-3-4-deprecate-configuration-options) (that is the minimum Symfony version we support)

Reply:
So, in other words, version 3 will behave like both settings have been set to their default values?

Reply:
These 2 settings are here to allow unregistered these 2 providers. But when running on Symfony 3.4+, they are both obsoleted by the LazyProvider, and so get unregistered 100% of the time.

Thus, these 2 settings should never have been needed. The ContainerAwareProvider and the BuilderServiceProvider are both configured by compiler passes. So we know when building the container whether they are useful at runtime or no (if they don't contain any menu definition, they are useless). This is the reason why the LazyProvider does not have an equivalent config option (everything happens automatically).
The config option is useful for `builder_alias`, because it is convention-based, not configuration-based, and so we don't know the list of menus it can build (and so being able to manually disable it if your project does not use it is good).